prisma/migrations/20240803171237_setup_schema/migration.sql

```diff
@@ -0,0 +1,270 @@
+-- CreateEnum
+CREATE TYPE "TransactionType" AS ENUM ('DEPOSIT', 'PURCHASE');
+
+-- CreateEnum
+CREATE TYPE "Role" AS ENUM ('USER', 'INSTRUCTOR', 'ADMIN');
+
+-- CreateEnum
+CREATE TYPE "DifficultyLevel" AS ENUM ('BEGINNER', 'INTERMEDIATE', 'ADVANCED');
+
+-- CreateEnum
+CREATE TYPE "NotificationType" AS ENUM ('ASSIGNMENT_SUBMISSION', 'COURSE_PURCHASE');
+
+-- CreateTable
+CREATE TABLE "User" (
+    "id" TEXT NOT NULL,
+    "email" TEXT NOT NULL,
+    "password" TEXT NOT NULL,
+    "phoneNumber" TEXT NOT NULL,
+    "campus" TEXT NOT NULL,
+    "role" "Role" NOT NULL DEFAULT 'USER',
+    "createdAt" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updatedAt" TIMESTAMP(3) NOT NULL,
+    "balance" DOUBLE PRECISION NOT NULL DEFAULT 0.0,
+
+    CONSTRAINT "User_pkey" PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "Course" (
```

Comment:
Make sure to add necessary indexes for the requirement: `I can search for courses and filter results by categories, difficulty levels, and ratings`. Please use composite indexes for this case.